### DISCUSSION TITLE
Is it possible to disable API routes?

### DISCUSSION THREAD

**Discussion by wescopeland** [vercel/next.js#11785] [Help] [answered]
We currently have a back-end implemented with Java, but we're wanting to host the front-end using Next.js.

Is it possible to disable API routes so calls to `/api/...` are fallen through rather than caught by the Next.js server? I'm having a hard time finding anything on the docs about such a setting.

**Comment by dheeraj-p** [comment]
What if you don't have a `/api` directory at all?

**Reply by wescopeland** [reply]
Unfortunately, this just causes Next.js to throw errors about certain `api` routes not existing, which is certainly a sensible default!

**Comment by Janpot** [comment, marked as answer]
If you want to proxy all `/api/...` calls to another backend, you should be able to use rewrites for that: https://github.com/zeit/next.js/issues/9081

**Reply by wescopeland** [reply]
This is the solution we're going to look into, thank you very much for pointing us in this direction!

**Reply by japem** [reply]
I know it's 2 years later but... what about for local development? My browser blocks CORS requests when I do this. In production it's not a problem, since the same origin will be used for my file servers and (non-Next.js) API routes, but my browser will refuse to follow a redirect to a service running locally on a different port from Next.js.

**Comment by ingoclaro** [comment]
you could have a reverse proxy in front to do that routing for you. nginx is a good one, but anything that you are familiar with would work.
This would also avoid having the api requests pass trough next/node, freeing up the app resources for only frontend requests

**Comment by wescopeland** [comment]
Thanks all for your help. We're going to look into both the path rewrites and using Nginx as a reverse proxy.

**Comment by amsimoes** [comment]
@Janpot what happens if I rewrite requests to `source: "/:path*"` (instead of `"/api/..."`) to another backend?

Just because it would be easier to do some testing and switching around between environments.

**Reply by Janpot** [reply]
It should be possible to do so. Have you tried:
```js
{ source: "/api/:path*", destination: "https://another-backend/:path*" }
```

**Comment by themagicalmammal** [comment]
I have done this i am getting two problems with this.

1. (node:295543) [DEP0060] DeprecationWarning: The `util._extend` API is deprecated. Please use Object.assign() instead.
(Use `node --trace-deprecation ...` to show where the warning was created)

2. /api routing works fine but /api/<anything_else> completely fails

**Reply by themagicalmammal** [reply]
{
    source: '/api/:path*',
    destination: `${backendBaseUrl}/api/:path*`
}
